lib/saluki-components/src/sources/checks/builder/python.rs

```diff
@@ -0,0 +1,312 @@
+                                        }
+                                    }
+                                    project_root
+                                }
+                                Err(e) => {
+                                    error!("Failed to get current executable path: {}", e);
+                                    Path::new(".").to_path_buf()
+                                }
+                            };
+
+                            // Add paths relative to executable location
+                            let dist_path = project_root.join("dist");
+                            let checks_d_path = dist_path.join("checks.d");
+
+                            p.insert(0, dist_path.to_string_lossy().as_ref()).unwrap(); // common modules are shipped in the dist path directly or under the "checks/" sub-dir
+                            p.insert(0, checks_d_path.to_string_lossy().as_ref()).unwrap(); // integrations-core legacy checks
+                            p.insert(0, "/etc/datadog-agent/checks.d/").unwrap(); // Agent checks folder
+
+                            info!("Python sys.path is: {:?}", p);
+                        }
+                        Err(e) => {
+                            return Err(generic_error!("Could not get sys.path {:?}", e));
+                        }
+                    };
+                }
+            } else {
+                return Err(generic_error!("Could not import sys module"));
+            }
+
+            // Validate that python env is correctly configured
+            let modd = match py.import("datadog_checks.checks") {
+                Ok(m) => m,
+                Err(e) => {
+                    if let Some(traceback) = e.traceback(py) {
+                        error!("Traceback: {}", traceback.format().expect("Could format traceback"));
+                    }
+                    return Err(generic_error!("Could not import datadog_checks module {:?}", e));
+                }
+            };
+            match modd.getattr("AgentCheck") {
+                Ok(c) => {
+                    info!("All pre-requisites for running python checks.");
+                    Ok(c.unbind())
+                }
+                Err(e) => {
+                    if let Some(traceback) = e.traceback(py) {
+                        error!("Traceback: {}", traceback.format().expect("Could format traceback"));
+                    }
+                    Err(generic_error!("Could not get AgentCheck class {:?}", e))
+                }
+            }
+        });
+
+        match result {
+            Ok(agent_check_class) => Self {
+                valid: true,
+                agent_check_class: Some(agent_check_class),
+            },
+            Err(e) => {
+                warn!("Could not load Python runtime. Cannot build Python check. {:?}", e);
+                Self {
+                    valid: false,
+                    agent_check_class: None,
+                }
+            }
+        }
```

Comment:
I'm still not convinced that this is the right pattern.

Why are we allowing execution to continue if the Python interpreter/environment fails to initialize properly?

Reply:
> Why are we allowing execution to continue if the Python interpreter/environment fails to initialize properly?

That is how the Go Agent works; if it fails to load the Python runtime, it does not crash, it simply continues. In this case, if we were to have other check builders we would like to try to build the check with those builders.

Reply:
Fair enough.